src/reduced_basis/rb_eim_construction.C

```diff
@@ -404,8 +406,10 @@ void RBEIMConstruction::set_rb_construction_parameters(unsigned int n_training_s
       const std::string & lookup_table_param_name =
         get_rb_eim_evaluation().get_parametrized_function().lookup_table_param_name;
 
-      std::vector<Real> lookup_table_training_samples(n_training_samples_in);
-      std::iota(lookup_table_training_samples.begin(), lookup_table_training_samples.end(), 0);
+      Real val = 0.0;
+      std::vector<RBParameter> lookup_table_training_samples(n_training_samples_in, {val});
+      for(auto it = lookup_table_training_samples.begin(); it != lookup_table_training_samples.end(); ++it, ++val)
+        (*it)[0] = val;
```

Comment:
I was a bit thrown off seeing that `val` is a `Real` and then seeing the `++val` however I see now that this is also what we were doing before behind the scenes with `std::iota`. Maybe add a comment here just to make it clear what's going on?
```
// Fill lookup_table_training_samples with the sequential single-entry vectors {0.0}, {1.0}, {2.0}, ...
```

Reply:
Yeah, it's a good comment. I've changed the code to be explicit about adding 1.0 instead, and added a comment.